test/test_generated_mypy.py

```diff
@@ -310,6 +320,17 @@ def test_which_oneof_proto3():
         s_untyped.WhichOneof("garbage")
 
 
+def test_extensions_proto2():
+    # type: () -> None
+    s = Simple1()
+
+    e1 = s.Extensions[Extensions1.ext]  # type: Extensions1
```

Comment:
May want to consider mypy-typing the `Extensions` variable on all messages to be a
```
Mapping[FieldDescriptor, google.protobuf.Message]
```

I'd also appreciate some kind of `assert isinstance(Extensions1.ext, FieldDescriptor)` to confirm that we're mypying it  to the correct type.

Reply:
I figured out a way to improve type inference for extensions.  Now we type `e1` as `Extensions1` explicitly.  Doing so required overwriting the `Extensions` defined for `Message`s in the typeshed.
I also added such `assert` statements.